src/python/pants/goal/stats_aggregator.py

```diff
@@ -55,13 +56,30 @@ class StatsAggregatorSubsystem(Subsystem):
         ),
         advanced=True,
     )
+    output_file = StrOption(
+        default=None,
+        metavar="<path>",
+        help="Output the stats to this file. If unspecified, outputs to stdout.",
+    )
+
+
+def log_or_write_to_file(output_file: Optional[str], text: str, logger: logging.Logger) -> None:
```

Comment:
what's the purpose of taking `logger` as an argument here? (it also shadows the global `logger` instance, so suggest to either remove the arg and use that, or rename the arg to not shadow it.)

could also consider marking this helper method as private with a leading underscore.

Reply:
and, consider what to do if `text` is empty.

Reply:
> what's the purpose of taking logger as an argument here? (it also shadows the global logger instance, so suggest to either remove the arg and use that, or rename the arg to not shadow it.)

I've just realized it's accessible globally, so removed the argument.

> and, consider what to do if text is empty.

Perhaps it doesn't make sense to log / write empty file, so will just skip doing anything!

Reply:
> could also consider marking this helper method as private with a leading underscore.

Done!